galley/pkg/config/analysis/analyzer.go

```diff
@@ -14,41 +14,89 @@
 
 package analysis
 
-import "istio.io/istio/galley/pkg/config/scope"
+import (
+	"istio.io/istio/galley/pkg/config/collection"
+	"istio.io/istio/galley/pkg/config/scope"
+)
 
 // Analyzer is an interface for analyzing configuration.
 type Analyzer interface {
-	Name() string
+	Metadata() Metadata
 	Analyze(c Context)
 }
 
+// Metadata represents metadata for an analyzer
+type Metadata struct {
+	name            string
+	collectionsUsed map[collection.Name]bool
+}
+
+// NewMetadata creates a new Metadata object
+func NewMetadata(name string, collectionsUsed collection.Names) Metadata {
+	cu := make(map[collection.Name]bool)
+	for _, c := range collectionsUsed {
+		cu[c] = true
+	}
+	return Metadata{
+		name:            name,
+		collectionsUsed: cu,
+	}
+}
+
+// Name of the analyzer this metadata is for
+func (m Metadata) Name() string {
+	return m.name
+}
+
+// CollectionsUsed registers the collections accessed by the analyzer this metadata is for
+func (m Metadata) CollectionsUsed() map[collection.Name]bool {
```

Comment:
Minor nit: can we call this "RequiredCollections", or simply "Inputs"?

Reply:
Inputs sounds fine.